### ISSUE TITLE
[AWS] Disable Public IP's creation per default

### ISSUE THREAD

**Issue by toschneck** [kubermatic/kubeone#635]
Since https://github.com/kubermatic/machine-controller/issues/544 we could disable the assignment of public IP's at the machine deployment. To ensure that the flag will set on every machine deployment, there should be an option to specify the `assignPublicIP: false`  in the `KubeOneCluster` config as enforced default.

cross ref: https://github.com/kubermatic/customer-projects/issues/1

**Comment by xmudrii** [comment]
We already support disabling assigning public IPs on MachineDeployments. You can see the `aws-private` Terraform example as a reference. I'm not sure should we make it default, so I'd like to discuss it first.

**Comment by toschneck** [comment]
@xmudrii  good to know, perfect! It shouldn't by the default in any case, but flag what ensures the machinecontroller enforces the "assingnment of private IPs". 

Anyway I wrote a small sh controller for enforcing to set the flag. So no pressure on this.

**Comment by kron4eg** [comment]
This should be a machine-controller flag, on a webhook for example.

**Comment by mariotti** [comment]
It should default to false. No public IP.

I write as a "user". In all the examples and demos I see setting a public IP, which would work for a web server, local tests, demos. But I start to think that most of the applications might be internal services.

For example I am setting up a jenkins server, which has to be in an internal network.
If it is specific to AWS, then it has to be internal, no elastic IP, no public IP, etc.

I think the problem rises when we think about: fully Kubernetes controlled? Or part is back
to the administration [AWS]? Some legacy services... or.. whatever..

**Comment by mariotti** [comment]
What about: https://github.com/kubermatic/kubeone/tree/master/examples/terraform/aws-private
Is it still really "experimental"?

**Comment by mariotti** [comment]
Beside a general security issue. Not related to k8s or aws, but we should never default to be open to the public. It is like the default password for the raspberry pi ssh.

The user should take a conscious action to open anything.

**Comment by kron4eg** [comment]
the only "experimental" part that's left in aws-private config are IAM permissions (which are too high for control plane nodes), besides that — it's OK.

**Comment by kron4eg** [comment]
Is this issue still relevant?

**Comment by kdomanski** [comment]
Yes

**Comment by kron4eg** [comment]
Then it should be moved to the machine-controller repository, the only place where `assignPublicIP: false` can be enforced is machine-controller's webhook.

**Comment by xmudrii** [comment]
@kron4eg @kdomanski I agree with Artiom that this issue is for `machine-controller`. I also believe that the best approach would be to extend the `machine-controller-webhook`. I propose that we close this issue.

**Comment by kron4eg** [comment]
We can keep this open and close it with PR integrating change from machine-controller (I suppose there will be some kind of flag).

But main task anyway doesn't belongs here.
